src/Components/Debugger.fs

```diff
@@ -336,34 +336,39 @@ module Debugger =
-                                      msbuildTasks
-                                      |> Seq.tryFind (fun t ->
-                                          t.group = Some vscode.TaskGroup.Build && t.name = projectFile)
-                                  // emit configurations for any launchsettings for this project
-                                  match readSettingsForProject p with
-                                  | Some launchSettings ->
-                                      yield! configsForProject (p, launchSettings, buildTaskForProject)
-                                  | None -> ()
-                                  // emit a default configuration for this project if it is an executable
-                                  match defaultConfigForProject (p, buildTaskForProject) with
-                                  | Some p -> yield p
-                                  | None -> () ])
-
-                        return ResizeArray tasks
-                    }
-
-                generate () // this bix/unbox is a hack because JS types
+                    match CSharpExtension.tryFindCSharpExtension () with
+                    | false ->
+                        CSharpExtension.warnAboutMissingCSharpExtension ()
+                        promise { return ResizeArray() }
+                    | true ->
+                        promise {
+                            logger.Info("Evaluating launch settings configurations for %O", folder)
+                            let projects = Project.getLoaded ()
+                            let! msbuildTasks = tasks.fetchTasks (msbuildTasksFilter)
+
+                            let tasks =
+                                projects
+                                |> List.collect (fun (p: Project) ->
+                                    [ let projectFile = node.path.basename p.Project
+
+                                      let buildTaskForProject =
+                                          msbuildTasks
+                                          |> Seq.tryFind (fun t ->
+                                              t.group = Some vscode.TaskGroup.Build && t.name = projectFile)
+                                      // emit configurations for any launchsettings for this project
+                                      match readSettingsForProject p with
+                                      | Some launchSettings ->
+                                          yield! configsForProject (p, launchSettings, buildTaskForProject)
+                                      | None -> ()
+                                      // emit a default configuration for this project if it is an executable
+                                      match defaultConfigForProject (p, buildTaskForProject) with
+                                      | Some p -> yield p
+                                      | None -> () ])
+
+                            return ResizeArray tasks
+                        }
```

Comment:
This is all the same logic, just wrapped in a 'if debugging exists then .. else' block